packages/api/pages/api/languages/[code]/verses/[verseId]/glosses.ts

```diff
@@ -37,28 +37,28 @@ export default createRoute<{ code: string; verseId: string }>()
         		JOIN "Word" ON "Word"."formId" = "VerseWord"."formId"
         		JOIN "Gloss" ON "Word"."id" = "Gloss"."wordId"
         			AND "Gloss"."languageId" = ${language.id}::uuid
-              AND "Gloss"."state" = 'APPROVED'
+        			AND "Gloss"."state" = 'APPROVED'
         		GROUP BY "VerseWord"."id", "Gloss"."gloss"
         	) AS "WordSuggestion"
         	GROUP BY "id"
         )
         -- Now we can gather the suggestions and other data for each word in the verse.
-        SELECT "VerseWord"."id" as "wordId", "Gloss"."gloss", "Suggestion"."suggestions", "Gloss"."state" FROM "VerseWord"
+        SELECT
+          "VerseWord"."id" as "wordId",
+          COALESCE("Gloss"."gloss", '') AS "gloss",
+          COALESCE("Suggestion"."suggestions", '{}') AS "suggestions",
+          COALESCE("Gloss"."state", 'UNAPPROVED') AS "state"
+        FROM "VerseWord"
         LEFT OUTER JOIN "Suggestion" ON "VerseWord"."id" = "Suggestion"."id"
-        JOIN "Gloss" ON "VerseWord"."id" = "wordId"
-        WHERE "Gloss"."languageId" = ${language.id}::uuid 
+        LEFT OUTER JOIN "Gloss" ON "VerseWord"."id" = "wordId"
+          AND "Gloss"."languageId" = ${language.id}::uuid
```

Comment:
I missed two things that ensure that an array of words is always returned if the verse exists even if there are no glosses for the verse.
* I needed to use a LEFT OUTER JOIN for glosses in the case that they don't exist.
* And then I needed to move the WHERE into the join so that it only governs which glosses are joined rather than which rows are returned in the results.